src/logic/wallets/__tests__/network.test.ts

```diff
@@ -77,32 +77,70 @@ describe('src/logic/wallets/utils/network', () => {
     })
   })
   describe('shouldSwitchNetwork', () => {
-    it('should return true when networks mismatch', () => {
-      const wallet = {
-        provider: {
-          networkVersion: '4',
-        },
-      }
-
-      expect(shouldSwitchNetwork(wallet as Wallet)).toBe(true)
-    })
-
     it('should return false when wallet is not connected', () => {
       const wallet = {
         provider: undefined,
       }
 
       expect(shouldSwitchNetwork(wallet as Wallet)).toBe(false)
     })
-
-    it('should return false when networks are the same', () => {
-      const wallet = {
-        provider: {
-          networkVersion: '1',
-        },
-      }
-
-      expect(shouldSwitchNetwork(wallet as Wallet)).toBe(false)
+    describe('should return true when networks mismatch', () => {
```

Comment:
I suggest setting `chainId` here to avoid possible dependency issues i.e.

`jest.spyOn(config, '_getChainId').mockImplementation(() => '4')`

Reply:
It is mocked at the top of the test.